test/gtest/ucp/test_ucp_memheap.cc

```diff
@@ -7,248 +7,97 @@
 
 #include "test_ucp_memheap.h"
 
+#include <common/mem_buffer.h>
 #include <common/test_helpers.h>
 #include <ucs/sys/sys.h>
 
 
-std::vector<ucp_test_param>
-test_ucp_memheap::enum_test_params(const ucp_params_t& ctx_params,
-                                   const std::string& name,
-                                   const std::string& test_case_name,
-                                   const std::string& tls)
+void test_ucp_memheap::init()
 {
-    std::vector<ucp_test_param> result;
-    generate_test_params_variant(ctx_params, name,
-                                 test_case_name, tls, 0, result);
-    generate_test_params_variant(ctx_params, name,
-                                 test_case_name + "/map_nb",
-                                 tls, UCP_MEM_MAP_NONBLOCK, result);
-    return result;
+    ucp_test::init();
+    sender().connect(&receiver(), get_ep_params());
 }
 
-void test_ucp_memheap::test_nonblocking_implicit_stream_xfer(nonblocking_send_func_t send,
-                                                             size_t size, int max_iter,
-                                                             size_t alignment,
-                                                             bool malloc_allocate,
-                                                             bool is_ep_flush)
+void test_ucp_memheap::test_xfer(send_func_t send_func, size_t size,
+                                 unsigned num_iters, size_t alignment,
+                                 ucs_memory_type_t send_mem_type,
+                                 ucs_memory_type_t target_mem_type,
+                                 unsigned mem_map_flags,
+                                 bool is_ep_flush, void *arg)
 {
-    void *memheap;
-    size_t memheap_size;
     ucp_mem_map_params_t params;
-    ucp_mem_attr_t mem_attr;
     ucs_status_t status;
+    ptrdiff_t padding;
+    ucp_mem_h memh;
 
-    memheap = NULL;
-    memheap_size = max_iter * size + alignment;
-
-    if (max_iter == DEFAULT_ITERS) {
-        max_iter = 300 / ucs::test_time_multiplier();
-    }
-
-    if (size == DEFAULT_SIZE) {
-        size = ucs_max((size_t)ucs::rand() % (12 * UCS_KBYTE), alignment);
-    }
-    memheap_size = max_iter * size + alignment;
+    ucs_assert(!(mem_map_flags & (UCP_MEM_MAP_ALLOCATE | UCP_MEM_MAP_FIXED)));
 
-    sender().connect(&receiver(), get_ep_params());
+    mem_buffer memheap(num_iters * size + alignment, target_mem_type);
+    padding = UCS_PTR_BYTE_DIFF(memheap.ptr(),
+                                ucs_align_up_pow2_ptr(memheap.ptr(), alignment));
+    ucs_assert(padding >= 0);
+    ucs_assert(padding < alignment);
 
+    /* Allocate heap */
```

Comment:
should we start all comments with lowercase letters?

Reply:
IMO better start with uppercase